### ISSUE TITLE
Accessing "SDL_Window" and "SDL_Renderer" Still w/ SDL-GPU

### ISSUE THREAD

**Issue by xgibbousx** [grimfang4/sdl-gpu#77]
Hi,

I am trying to integrate LibRocket (GUI library) into my engine that uses SDL-GPU.  They have code that integrates with SDL2 but it uses SDL_Window, SDL_Renderer, SDL_SetRenderDrawBlendMode, SDL_RenderDrawPoint, etc. in its implementation.

SDL-GPU seems to replace the SDL_Renderer and SDL_Window with a GPU_Target.

Is there a way to still get access to SDL_Window and SDL_Renderer to be used with SDL_SetRenderDrawBlendMode / SDL_RenderDrawPoint while using SDL-GPU?

Thank you!

**Comment by xgibbousx** [comment]
Decided to use a different GUI library because I believe LibRocket doesn't support an OpenGL 3+ programmable pipeline.  Going to try and use dear imgui instead which definitely has OpenGL 3+ examples.

**Comment by grimfang4** [comment]
Integrating SDL_gpu with librocket takes a bit of work, but it certainly should be possible.  I haven't done it myself, but the rendering part would primarily use GPU_TriangleBatch().  Librocket will generate the triangle data (vertices, indices, texture coordinates...) that you need to render.  See interface docs here:
http://librocket.com/wiki/documentation/C%2B%2BManual/Interfaces#Therenderinterface

On the other hand, I already have some code to work with dear imgui that is not cleaned up yet, but just ask and I'll send it over.  When I do clean it up, it'll go into the SDL_gpu repo and I'll submit a pull request to imgui.

**Comment by xgibbousx** [comment]
Yes, I have decided to use dear imgui and would love the code you have!!!

Also, dear imgui needs access to the window so the questions above may still be valid!

Thanks so much!

**Comment by grimfang4** [comment]
Alright, sent it over.

The window associated with any context target (GPU_Target that has a non-NULL context member, i.e. it renders to a window instead of an image) can be gotten like so:
`SDL_Window* window = SDL_GetWindowFromID(target->context->windowID);`

Since SDL_gpu does replace SDL_Renderer, it is not advised to mix with SDL_Render* functions.  SDL_gpu calls can do everything that SDL_Renderer does and much more.  See GPU_SetShapeBlending(), GPU_SetShapeBlendMode(), and GPU_Pixel().

**Comment by xgibbousx** [comment]
Perfect!

I got everything working.  I ended up just using your window access code above along with the boilerplate imgui_impl_sdl_gl3.cpp provided in the imgui SDL2+OpenGL3  example code.  Everything worked perfectly the first time around which is amazing.

However, your other code helped as a comparison to see/understand everything that is going on.  Thank you for sending that over.

Thanks again so much!
